source/plugin/advanced/debugging.rst

```diff
@@ -0,0 +1,119 @@
+
+Eclipse
+~~~~~~~
+
+.. tip:: You can use `Setting Up Eclipse <../basics/workspace/eclipse.rst>`_ to create a new Eclipse project. Eclipse, unlike IntelliJ, does not require any specific setup in regards to how you manage the plugin. Creating a new project is all you need to do.
+
+Adding Plugin to Sponge classpath
+=================================
+
+The idea behind this is that we'll launch Sponge from within your IDE, like normal. However, the difference is that
+we'll be adding your plugin to the classpath. To do this should only take a few steps, and will enhance your workflow,
+having cut down on all of the time that you previously would have spent transferring files and manually starting your
+server.
+
+.. tip:: Please ensure that you have your Run/Debug Configuration(s) set appropriately, as shown in the Sponge `README.md <https://github.com/SpongePowered/Sponge/blob/master/README.md>`_
+
+IntelliJ
+~~~~~~~~
+
+* Open your Project Structure.
+
+  * Click ``File``, followed by ``Project Structure...``.
+  * OR, click the ``Project Structure`` icon, in the upper right-hand corner of the IDE, next to the ``Search`` icon.
+
+* Click ``Modules``.
+* Click the Green ``+`` symbol (``Add``), and select ``Module Dependency``
+* Select your Module (Plugin).
+* Do NOT check the module, after it is added to the list.
+
+Eclipse
+~~~~~~~
+
+* Find your Run/Debug Configuration
+
+  * Click ``Run``, followed by ``Run Configurations...``
+  * OR, click the drop-down arrow beside the Run/Debug icons and then ``Run Configurations...`` or ``Debug Configurations...``, respectively.
+
+* Select your Run/Debug Configuration for Sponge (Server) on the left side.
+* Switch to the ``Classpath`` tab.
+* Select ``User Entries``, followed by the ``Add Projects...`` button.
+* Select the appropriate Project for your Plugin.
+* Click the ``OK`` button.
+* Click the ``Apply`` button on the bottom right corner.
+
+Testing the Configuration
+=========================
+
+After you've followed the previous steps, you should be ready to start debugging.
+
+IntelliJ
+~~~~~~~~
+
+Rather than pressing the Green right-pointing arrow to run your Run/Debug configuration, click the Green icon to the
+right of it, ``Debug``. This is to prepare for Hotswapping, which we're about to cover.
+
+Eclipse
+~~~~~~~
+
+Rather than pressing the green right-pointing arrow to run your Run/Debug configuration, click the drop down arrow of
+the Debug icon (the one displaying a bug) and click your Sponge (Server) configuration. If it doesn't appear in the
+drop-down menu, click ``Debug Configurations``. Select ``Sponge (Server)`` configuration and hit the ``Debug`` button
+on the bottom left.
+
+Hotswapping
+===========
+
+Now that you've got the server, running, let's try to make some changes. Note, that there are limits to the built-in
+capabilities of Hotswapping:
+
+* You cannot create or remove methods, nor can you create or remove classes.
```

Comment:
You can't change fields, method parameters/return types etc too. You can create classes though, because they are loaded when first called.